src/code/z_parameter.c

```diff
@@ -2239,111 +2239,126 @@ void Inventory_ChangeAmmo(s16 item, s16 ammoChange) {
 }
 
 void Magic_Fill(GlobalContext* globalCtx) {
-    if (gSaveContext.magicAcquired) {
-        gSaveContext.unk_13F2 = gSaveContext.unk_13F0;
-        gSaveContext.unk_13F6 = (gSaveContext.doubleMagic * 0x30) + 0x30;
-        gSaveContext.unk_13F0 = 9;
+    if (gSaveContext.isMagicAcquired) {
+        gSaveContext.magicBarActionStored = gSaveContext.magicBarAction;
+        gSaveContext.magicCapacity = (gSaveContext.isDoubleMagicAcquired + 1) * MAGIC_HALF_BAR;
+        gSaveContext.magicBarAction = MAGIC_BAR_ACTION_FILL;
     }
 }
 
-void func_800876C8(GlobalContext* globalCtx) {
-    if ((gSaveContext.unk_13F0 != 8) && (gSaveContext.unk_13F0 != 9)) {
-        if (gSaveContext.unk_13F0 == 10) {
-            gSaveContext.unk_13F2 = gSaveContext.unk_13F0;
+void Magic_ResetMagicBarAction(GlobalContext* globalCtx) {
+    if ((gSaveContext.magicBarAction != MAGIC_BAR_ACTION_GROW_WIDE) &&
+        (gSaveContext.magicBarAction != MAGIC_BAR_ACTION_FILL)) {
+        if (gSaveContext.magicBarAction == MAGIC_BAR_ACTION_ADD) {
+            gSaveContext.magicBarActionStored = gSaveContext.magicBarAction;
         }
-        gSaveContext.unk_13F0 = 5;
+        gSaveContext.magicBarAction = MAGIC_BAR_ACTION_RESTORE_IDLE;
     }
 }
 
-s32 func_80087708(GlobalContext* globalCtx, s16 arg1, s16 arg2) {
-    if (!gSaveContext.magicAcquired) {
-        return 0;
+s32 Magic_ChangeBy(GlobalContext* globalCtx, s16 magicChange, s16 changeType) {
```

Comment:
I was actually inspired by `Health_ChangeBy` when naming this function. I acknowledge the implementation details are slightly different, and it might be worth adding a comment above both functions emphasizing the different nature of the `XXXChange` arg1 required by each.

Maybe something like `Magic_ChangeByTarget` to emphasize that this function doesn't change magic directly, but sets a target for `Interface_UpdateMagicBar` to act upon?

Reply:
What about `Magic_RequestChange`?

Reply:
I guess we expect setup functions to not be able to fail (like, this function can return false if not enough magic for example), so I guess `Magic_RequestChange` is a good idea

Reply:
request sounds good
I was never a fan of the `healthChange` arg name, and would probably like to see it changed for both health and magic. even just adding the word "amount" makes it much better, the amount to change by

Reply:
interestingly you can also argue that Health_ChangeBy is a request cause it will fail and return false when health is already less than or equal to 0. But its less impactful than the amount of conditions that magic changing can fail under

Reply:
Could it just be 
`magicChange` -> `amount`
`changeType` -> `type`
and it's implied? The only thing is to ensure `amount` for magic is always positive since increasing or decreasing is determined by type, which is different than Health in which you use use either a positive or negative value

Reply:
that could work, it could also be `changeAmount` and `changeType`  for maximum clarity 
health could also use `changeAmount`

that note about negative magic values is def important for a function level comment

Reply:
id actually be fine with `amount` and `type` tbh

Reply:
actually wait, `amount` could be interpreted to mean a new value to set to as opposed and amount to change by?

Reply:
Ah so you don't want it to be interpreted as a target instead of a change